--- a/frontend/frontend_base.h
+++ b/frontend/frontend_base.h
@@ -443,8 +443,11 @@
     // Import a leaf cell - (white|black)box
     void import_leaf_cell(HierModuleState &m, const std::string &name, const cell_dat_t &cd)
     {
-        IdString inst_name = unique_name(m.prefix, name, false);
+        auto cell_type = impl.get_cell_type(cd);
+        if (cell_type == "$scopeinfo")
+            return;
+        IdString inst_name = unique_name(m.prefix, name, false);
         ctx->hierarchy[m.path].leaf_cells_by_gname[inst_name] = ctx->id(name);
         ctx->hierarchy[m.path].leaf_cells[ctx->id(name)] = inst_name;
-        CellInfo *ci = ctx->createCell(inst_name, ctx->id(impl.get_cell_type(cd)));
+        CellInfo *ci = ctx->createCell(inst_name, ctx->id(cell_type));
         ci->hierpath = m.path;
